src/possible_moves.c

```diff
@@ -0,0 +1,513 @@
+#include "possible_moves.h"
+
+/*4 func to cal if peice is getting off board in hes move*/
+char pass_left(char src){
+    if(get_column(src) == 0)
+        return 1;
+    return 0;
+}
+
+
+char pass_right(char src){
+    if(get_column(src) == 7)
+        return 1;
+    return 0;
+}
+
+
+char pass_down(char src){
+    if(get_row(src) == 0)
+        return 1;
+    return 0;
+}
+
+
+char pass_up(char src){
+    if(get_row(src) == 7)
+        return 1;
+    return 0;
+}
+
+/*check if peice will get on a square that an ally is on which means he can't*/
+char colid_with_ally(char dst, board the_board, char color)
+{
+    char dst_piece = get_piece_in_square(&the_board, dst);
+    if(color)
+    {
+        if(dst_piece <= 6 && dst_piece != 0)
+            return 1;
+        return 0;
+    }
+    if(dst_piece > 6)
+        return 1;
+    return 0;
+}
+
+/*check if peice is on a square that an enemy is on which means he can't continue*/
+char colid_with_enemy(char src, board the_board, char color){
+    char src_piece = get_piece_in_square(&the_board, src);
+    if(color)
+    {
+        if(src_piece > 6)
+            return 1;
+        return 0;
+    }
+    if(src_piece <= 6 && src_piece != 0)
+        return 1;
+    return 0;
+}
+
+/*3 func to check if the pawn by his color can move and if so to where*/
+char pawn_straight(char src, board the_board, char color){
+    if(color){
+        if(get_piece_in_square(&the_board, src + UP) == 0)
+            return 1;
+        return 0;
+    }
+    if(get_piece_in_square(&the_board, src + DOWN) == 0)
+            return 1;
+        return 0;
+}
+
+
+char pawn_eat_left(char src, board the_board, char color){
+    if(!pass_left(src)){
+        if(color)
+            return colid_with_enemy(src + UP_LEFT, the_board, color);
+        return colid_with_enemy(src + DOWN_LEFT, the_board, color);
+    }
+    return 0;
+}
+
+
+char pawn_eat_right(char src, board the_board, char color){
+    if(!pass_right(src)){
+        if(color)
+            return colid_with_enemy(src + UP_RIGHT, the_board, color);
+        return colid_with_enemy(src + DOWN_RIGHT, the_board, color);
+    }
+    return 0;
+}
+
+/*8 func to return if the peice can move to any of his 8 sides (if the peice itself can of course)*/
+char move_up(char src, board the_board, char color){
+    if(colid_with_enemy(src, the_board, color) || pass_up(src))
```

Comment:
write     if(colid_with_enemy(src, the_board, color) || pass_up(src) || colid_with_ally(src + UP, the_board, color))
do it in all 8 function.